elasticdl/python/master/embedding_service.py

```diff
@@ -65,21 +58,33 @@ def start_embedding_service(
         restart_policy="Never",
         **kargs,
     ):
-        self.start_embedding_pod_and_redis(
-            command=["python"],
-            args=["-m", "elasticdl.python.master.embedding_service"],
-            embedding_service_id=embedding_service_id,
-            resource_request=resource_request,
-            resource_limit=resource_limit,
+        logger.info("Starting pod for embedding service ...")
+        k8s_client = k8s.Client(**kargs)
+        pod = k8s_client.create_embedding_service(
+            worker_id=embedding_service_id,
+            resource_requests=resource_request,
+            resource_limits=resource_limit,
             pod_priority=pod_priority,
             volume=volume,
             image_pull_policy=image_pull_policy,
+            command=command,
+            args=args,
             restart_policy=restart_policy,
-            **kargs,
         )
-        return self._create_redis_cluster()
 
-    def _create_redis_cluster(self):
+        # TODO: assign address with pod's domain name instead of pod's ip.
+        # and should not use fixed ports
+        address_ip = pod.status.pod_ip
+        while not address_ip:
+            pod = k8s_client.get_embedding_service_pod(embedding_service_id)
```

Comment:
Is it better we wait for some time before querying again?

Reply:
Not very sure, just follow the previous code logic.